### PR TITLE
Service agent

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,43 +43,83 @@ public
...
 ## Agent
 
-### Getting and running the agent
+### Install
 
 Agent binaries can be found under wiresteward releases:
 https://github.com/utilitywarehouse/wiresteward/releases
 
-to install on linux simply:
+To install on linux simply:
```
Comment (george-angel) [inline README.md:51]: Since we are changing README, can we remove all the "simply"? Its redundant at best and at worst patronising.

Comment (ffilippopoulos) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -39,43 +43,83 @@ public
...
 
+- creating new network itun devices
+- Generating and posting wireguard keys to the server
+- Fetching oauth tokens to pass server authentication
+- Configuring wireguard peers
+- Configuring routes for the subnets allowed by the server
 thus it needs NET_ADMIN capabilities.
 
-To run simply:
+To try it one can simply:
```
Comment (george-angel) [inline README.md:68]: More "simply" ))

File: agent_http.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		if err != nil {
+			fmt.Fprintf(
+				w,
+				"error fetching token from web: %v",
+				err,
+			)
+			return
+		}
+		agentLeaseLoop(agentConf, token.IdToken)
+		fmt.Fprintf(w, "finished running agent! you can close this window now")
```
Comment (george-angel) [inline agent_http.go:22]: Agent is still running? "Auth is now complete"?

File: README.md [hunk 4]
```diff
@@ -39,43 +43,83 @@ public
...
+```
+
+### Run
 
-Agent takes a config file as an argument (or `~/.wiresteward.json` if not
-specified), from where it gets all the details needed to get a token from okta
-and create/update the wg interfaces and routes after talking to the remote
-wiresteward peers.
+The agent runs a local server on port 8080 and expects the user to visit
+`localhost:8080/renew` to trigger all the actions described above (under #Usage)
```
Comment (george-angel) [inline README.md:106]: 8080

File: README.md [hunk 5]
```diff
@@ -39,43 +43,83 @@ public
...
+
+Opening `localhost:8080/renew` on a browser will trigger the oauth process, if
```
Comment (george-angel) [inline README.md:108]: 8080

File: agent_http.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+					"error creating web url to get token: %v",
+					err,
+				)
+				return
+			}
+			http.Redirect(w, r, url, 302)
+			return
+		}
+		agentLeaseLoop(agentConf, token.IdToken)
+		fmt.Fprintf(w, "Agent refreshed and running! you can close this window now")
```
Comment (george-angel) [inline agent_http.go:43]: Why is half the app using `log` and the other half is using `fmt`?

Comment (ffilippopoulos) [reply agent_http.go:43]: I am using fmt when writing to `http.ResponseWriter` here, and log when the app is logging if I understand the question right? I updated everything that is an up log to use `log`

Comment (george-angel) [reply agent_http.go:43]: Wrong place for a comment, but yes, everywhere else where we are writing to stdout

Comment (george-angel) [reply agent_http.go:43]: Sorry to be picky, but if you are going to use punctuation then you need to capitalise the next word ))

**Pull request merged**